Upload seems to be broken since v0.4.1

I've tested the upload target `temporary-public-storage` as well as `filesystem` both with the same result. At least `temporary-public-storage` is working with lhci <= v0.4.0: 

```
lhci autorun --upload.target=temporary-public-storage
[...]
TypeError: Cannot read property 'trim' of null
    at getCurrentBranchRaw_ (/usr/local/lib/node_modules/@lhci/cli/node_modules/@lhci/utils/src/build-context.js:127:32)
    at getCurrentBranch (/usr/local/lib/node_modules/@lhci/cli/node_modules/@lhci/utils/src/build-context.js:139:18)
    at Object.buildCommand (/usr/local/lib/node_modules/@lhci/cli/src/upload/upload.js:111:16)
    at Object.builder (/usr/local/lib/node_modules/@lhci/cli/src/cli.js:56:17)
    at Object.runCommand (/usr/local/lib/node_modules/@lhci/cli/node_modules/yargs/lib/command.js:186:35)
    at Object.parseArgs [as _parseArgs] (/usr/local/lib/node_modules/@lhci/cli/node_modules/yargs/yargs.js:1063:30)
    at Object.get [as argv] (/usr/local/lib/node_modules/@lhci/cli/node_modules/yargs/yargs.js:1004:21)
    at run (/usr/local/lib/node_modules/@lhci/cli/src/cli.js:82:45)
    at Object.<anonymous> (/usr/local/lib/node_modules/@lhci/cli/src/cli.js:120:1)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
```

Is it possible that the docker images patrickhulce/lhci-client:0.4.1 and 0.4.2 are missing git:
```
$ docker run patrickhulce/lhci-client:0.4.0 git --version
git version 2.11.0
```
```
$ docker run patrickhulce/lhci-client:0.4.1 git --version
internal/modules/cjs/loader.js:969
  throw err;
  ^

Error: Cannot find module '/home/lhci/reports/git'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

--- a/docs/recipes/docker-client/Dockerfile
+++ b/docs/recipes/docker-client/Dockerfile
@@ -1,7 +1,7 @@
 FROM node:12-buster-slim
 
 # Install utilities
-RUN apt-get update --fix-missing && apt-get -y upgrade && apt-get install -y wget gnupg && apt-get clean
+RUN apt-get update --fix-missing && apt-get -y upgrade && apt-get install -y git wget gnupg && apt-get clean
 
 # Install latest chrome stable package.
 RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -

--- a/packages/cli/src/upload/upload.js
+++ b/packages/cli/src/upload/upload.js
@@ -30,6 +30,7 @@ const {
   getAncestorHashForBase,
   getAncestorHashForBranch,
   getGitHubRepoSlug,
+  getCurrentBranchSafe,
 } = require('@lhci/utils/src/build-context.js');
 
 /** @param {string} message */
@@ -108,7 +109,7 @@ function buildCommand(yargs) {
       type: 'boolean',
       description:
         '[temporary-public-storage only] Whether to post links to historical base results to storage or not. Defaults to true only on master branch.',
-      default: getCurrentBranch() === 'master',
+      default: getCurrentBranchSafe() === 'master',
     },
     urlReplacementPatterns: {
       type: 'array',

--- a/packages/utils/src/build-context.js
+++ b/packages/utils/src/build-context.js
@@ -130,8 +130,8 @@ function getCurrentBranchRaw_() {
     encoding: 'utf8',
   });
 
-  const branch = result.stdout.trim();
-  if (result.status !== 0 || branch === 'HEAD') {
+  const branch = typeof result.stdout === 'string' && result.stdout.trim();
+  if (result.status !== 0 || !branch || branch === 'HEAD') {
     throw new Error(
       'Unable to determine current branch with `git rev-parse --abbrev-ref HEAD`. ' +
         'This can be overridden with setting LHCI_BUILD_CONTEXT__CURRENT_BRANCH env.'
@@ -142,12 +142,25 @@ function getCurrentBranchRaw_() {
 }
 
 /**
+ * Returns the current branch name. Throws if it could not be determined.
  * @return {string}
  */
 function getCurrentBranch() {
   const branch = getCurrentBranchRaw_();
   if (branch === 'HEAD') throw new Error('Unable to determine current branch');
   return branch.replace('refs/heads/', '').slice(0, 40);
+}
+
+/**
+ * Returns the current branch name. Returns `undefined` if it could not be determined.
+ * @return {string|undefined}
+ */
+function getCurrentBranchSafe() {
+  try {
+    return getCurrentBranch();
+  } catch (err) {
+    return undefined;
+  }
 }
 
 /**
@@ -335,6 +348,7 @@ module.exports = {
   getCurrentHash,
   getCommitTime,
   getCurrentBranch,
+  getCurrentBranchSafe,
   getExternalBuildUrl,
   getCommitMessage,
   getAuthor,

